docs/dev/concepts.rst

```diff
@@ -0,0 +1,175 @@
+            "type": "object",
+            "properties": {
+               "message": {
+                  "default": "",
+                  "description": "Message to log to the console",
+                  "type": "string",
+               }
+            },
+            "required": [
+               "message"
+            ],
+         }
+      },
+      "arguments": {
+         "message": "It works!"
+      }
+   }
+
+Note that the recipe as returned by the API also contains data about the action,
+including a schema describing the arguments field.
+
+.. _JEXL: https://github.com/TechnologyAdvice/Jexl
+.. _Telemetry: https://wiki.mozilla.org/Telemetry
+
+.. _runtime:
+
+Runtime
+-------
+The runtime is an execution environment that can affect a running instance of
+Firefox. Upon activation (typically a few moments after Firefox launches), the
+runtime:
+
+1. Downloads :ref:`recipes <recipes>` from the Normandy service.
+2. Evaluates the recipe filters and filters out recipes that do not match the
+   client the runtime is installed within.
+3. Downloads the :ref:`actions <actions>` for the remaining recipes.
+4. Executes the action code for each recipe in a sandbox, passing in the
+   arguments from the recipe, and a :doc:`driver object </dev/driver>` containing
+   methods that can perform privileged actions.
+
+The runtime is implemented as a system add-on in the normandy-addon_ repository.
+
+.. _normandy-addon: https://github.com/mozilla/normandy-addon
+
+Threat Model
+------------
+Since the goal of SHIELD is to allow Mozilla to perform certain privileged
+actions quickly without shipping full updates to Firefox, it is a tempting
+target for compromising Firefox users. Normandy includes several security
+controls to help mitigate this risk.
+
+Action and Recipe Signing
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Actions and recipes that are downloaded by the runtime are signed using the
+autograph_ service. The runtime verifies the signature upon downloading the
+recipes and actions, ensuring that the runtime only executes recipes that have
+been signed with a Mozilla-controlled key.
+
+This helps prevent Man-in-the-Middle attacks where an adversary pretends to be
+the remote Normandy service.
+
+.. _autograph: https://github.com/mozilla-services/autograph
+
+Peer Approval
+^^^^^^^^^^^^^
+Recipes cannot be enabled in the Normandy admin interface without going through
+an approval process. One user must submit the recipe for approval, and a
+separate user must approve the recipe before it can be distributed by the
+service.
+
+This helps prevent compromise of a single account from compromising the entire
+service, since two accounts need to be compromised to publish a recipe.
+
+Action Sandbox
+^^^^^^^^^^^^^^
+Actions are executed within a JavaScript sandbox by the runtime. The sandbox
+limits the access of the JavaScript to prevent it from modifying Firefox in ways
+that haven't been reviewed and approved beforehand.
```

Comment:
It would be interested to document what that approval process look like. Maybe there should be a `secreview+` from the seceng team when the sandbox changes? I don't know what's appropriate.